examples/flyin_plugin/flyin_plugin/vscode.py

```diff
@@ -0,0 +1,169 @@
+# %% [markdown]
+# # FlyIn Vscode Decorator
+#
+# `@vscode` converts a Python task into a Visual Studio Code server with just one line change,
+# enabling connection and debugging in remote environments.
+
+# %%
+from flytekit import task, workflow
+from flytekitplugins.flyin import vscode
+
+# %% [markdown]
+# ## Usage
+# ### 1. Adding vscode decorator
+
+# %%
+@task
+@vscode
+def train():
+    print("forward")
+    print("backward")
+
+
+@workflow
+def wf_train():
+    train()
+
+
+# %% [markdown]
+# The `@vscode` decorator, when applied, converts a task into a Visual Studio Code server during runtime. This process overrides the standard execution of the task's function body, initiating a command to start a Visual Studio Code server instead.
+# ### 2. Connecting to the VSCode Server
+# You have two methods for connecting:
+# 1. **(Recommended)** Set up ingress on the backend to expose a URL on the Flyte console. Details are to be determined (TBD).
+
+# 2. **Use Port-Forwarding:** Execute the command:
+#    ```
+#    $ kubectl port-forward <pod name> <port>
+#    ```
+#    Then, open a browser and navigate to `localhost:port`. You will be presented with the interface as shown in the image below.
+
+# ### 3. Interactively Debugging Task
+#
+# To run the task in VSCode, select "Run and debug" from the left panel and execute the "interactive debugging" configuration.
+#
+# This will run your task with inputs from the previous task. It's important to note that the task runs entirely within VSCode and does not write the output to Flyte storage.
+#
+# For inspecting intermediate states, set breakpoints in the Python code and use the debugger for tracing.
+#
+# The `launch.json` file generated by FlyIn simply offers a convenient method to run the task. You can still use VSCode as you would locally. For instance, you can run a Python script directly from the embedded terminal using `python hello.py`.
+#
+# <img src="https://raw.githubusercontent.com/ByronHsu/static-resources/byhsu/flyin/flytesnacks/integrations/flyin/interactive_debugging.png">
+
+
+# ### 4. Resuming Task with Updated Code:
+# After you finished debugging, you can resume the task with updated code by executing the "resume task" configuration.
+# This will **terminate the code server**, run the task with inputs from the previous task, and write the output to Flyte storage.
+#
+# ** Note: Please remember to persist your code somewhere before resuming the task because you will lose the connection to vscode afterwards.**
```

Comment:
```suggestion
# ** Note: Remember to persist your code locally before resuming the task, since you will lose the connection to the VSCode server afterwards.**
```

Reply:
This is not precise. They will mostly push their code GitHub

Reply:
changed to "# ** Note: Remember to persist your code (e.g. to GitHub) before resuming the task, since you will lose the connection to the VSCode server afterwards.**"